DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py

```diff
@@ -94,27 +97,34 @@
 	process.castorDigis.InputLabel = rawTag
 
 process.emulTPDigis = process.simHcalTriggerPrimitiveDigis.clone(
-   inputLabel = ["hcalDigis", 'hcalDigis'],
+   inputLabel = cms.VInputTag("hcalDigis", "hcalDigis:ZDC"),
    FrontEndFormatError = True,
    FG_threshold = 2,
    InputTagFEDRaw = rawTag,
    upgradeHF = True,
    upgradeHE = True,
    upgradeHB = True,
-   inputUpgradeLabel = ["hcalDigis", "hcalDigis"],
+   inputUpgradeLabel = cms.VInputTag("hcalDigis", "hcalDigis:ZDC"),
    # Enable ZS on emulated TPs, to match what is done in data
    RunZS = True,
    ZS_threshold = 0
 )
 
+#inserting zdc emulator after tp digis
+process.etSumZdcProducer = cms.EDProducer('L1TZDCProducer',
+                                          hcalTPDigis = cms.InputTag("emulTPDigis"),
+                                          bxFirst = cms.int32(-2),
+                                          bxLast = cms.int32(3)
+)
+
 process.hcalDigis.InputLabel = rawTag
 process.emulTPDigisNoTDCCut = process.emulTPDigis.clone(
      parameters = cms.untracked.PSet(
 	ADCThresholdHF = cms.uint32(255),
 	TDCMaskHF = cms.uint64(0xFFFFFFFFFFFFFFFF)
      )
 )
-process.HcalTPGCoderULUT.LUTGenerationMode = False
+process.HcalTPGCoderULUT.LUTGenerationMode = True
```

Comment:
what's the purpose of this change?

Reply:
This is actually what we would like feedback on from @denizsun and @salimcerci, this is necessary for the code to run with the candidate GT (which does not yet include an updated L1TriggerObject) , but probably not necessary for actual data-taking. In this case, should we include this or leave it out?